src/runtime/internal/style_manager.ts

```diff
@@ -2,13 +2,19 @@ import { append_empty_stylesheet, get_root_for_style } from './dom';
 import { raf } from './environment';
 
 interface ExtendedDoc extends Document {
-	__svelte_stylesheet: CSSStyleSheet;
-	__svelte_rules: Record<string, true>;
+	[key: string]: any;
```

Comment:
If you would write it like this, you will keep the type information:
```ts
interface ExtendedDoc extends Document {
	[svelte_stylesheet]: CSSStyleSheet;
	[svelte_rules]: Record<string, true>;
}

const svelte_stylesheet = Symbol(`__svelte_stylesheet_${unique_id}`);
const svelte_rules = Symbol(`__svelte_rules_${unique_id}`);
```

> Note: with TypeScript version > `4.1` it would be possible to type the interface without the usage of `Symbol`

Reply:
If a @maintainer thinks this produces to much code than please at least type the variable initializatoons
```ts
	const stylesheet: CSSStyleSheet = ...
	const current_rules: Record<string, true> = ...
```

Reply:
I'm definitely open to either change. This is certainly a style type discussion regarding the application of TS across the codebase; I don't have visibility into this particular concern. The interface in question is only used in this small section of code and not exported, so the latter of typing variable initialization is adequate IMHO.